### DISCUSSION TITLE
MooseMatView

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [idaholab/moose#23568] [Q&A General] [answered]
Hello All

Do you know if moose can output a matrix or a vector? If so, how to output?

Thanks

**Comment by dschwen** [comment]
Are you asking about material property output? You need to do that componentwise. adding `outputs = exodus`  in the material block might already do the trick.

**Reply by nanguaxiaofendui** [reply]
Thank you very much

**Comment by GiudGiud** [comment, marked as answer]
If you are asking for outputting the system matrix, you ll want to pass to petsc:  -ksp_view_pmat
See https://github.com/idaholab/moose/discussions/17089

**Reply by nanguaxiaofendui** [reply]
Thank you very much. I'll try.
